doc/api/brotli.md

```diff
@@ -0,0 +1,404 @@
+* `brotli.constants.BROTLI_MAX_CHUNK`
+* `brotli.constants.BROTLI_DEFAULT_CHUNK`
+
+Compression mode.
+
+* `brotli.constants.BROTLI_MODE_GENERIC`
+* `brotli.constants.BROTLI_MODE_TEXT`
+* `brotli.constants.BROTLI_MODE_FONT`
+* `brotli.constants.BROTLI_DEFAULT_MODE`
+
+Quality levels.
+
+* `brotli.constants.BROTLI_MIN_QUALITY`
+* `brotli.constants.BROTLI_MAX_QUALITY`
+* `brotli.constants.BROTLI_DEFAULT_QUALITY`
+
+LGWIN values.
+
+* `brotli.constants.BROTLI_MIN_WINDOW_BITS`
+* `brotli.constants.BROTLI_MAX_WINDOW_BITS`
+* `brotli.constants.BROTLI_LARGE_MAX_WINDOW_BITS`
+* `brotli.constants.BROTLI_DEFAULT_WINDOW`
+
+LGBLOCK values.
+
+* `brotli.constants.BROTLI_MIN_INPUT_BLOCK_BITS`
+* `brotli.constants.BROTLI_MAX_INPUT_BLOCK_BITS`
+
+NPOSTFIX values.
+
+* `brotli.constants.BROTLI_MAX_NPOSTFIX`
+
+Parameters.
+
+* `brotli.constants.BROTLI_PARAM_MODE`
+* `brotli.constants.BROTLI_PARAM_QUALITY`
+* `brotli.constants.BROTLI_PARAM_LGWIN`
+* `brotli.constants.BROTLI_PARAM_LGBLOCK`
+* `brotli.constants.BROTLI_PARAM_DISABLE_LITERAL_CONTEXT_MODELING`
+* `brotli.constants.BROTLI_PARAM_SIZE_HINT`
+* `brotli.constants.BROTLI_PARAM_LARGE_WINDOW`
+* `brotli.constants.BROTLI_PARAM_NPOSTFIX`
+* `brotli.constants.BROTLI_PARAM_NDIRECT`
+* `brotli.constants.BROTLI_DECODER_PARAM_DISABLE_RING_BUFFER_REALLOCATION`
+* `brotli.constants.BROTLI_DECODER_PARAM_LARGE_WINDOW`
+
+## Class: Options
+<!--type=misc-->
+
+Each class takes an `options` object. All options are optional.
+
+Note that some options are only relevant when compressing, and are
+ignored by the decompression classes.
+
+* `chunkSize` {integer}
+**Default:** `16 * 1024`
+* `mode` {integer} (compression only)
+**Default:** `brotli.constants.BROTLI_DEFAULT_MODE`
+* `quality` {integer} (compression only)
+**Default:** `brotli.constants.BROTLI_DEFAULT_QUALITY`
+* `lgwin` {integer} (compression only)
+**Default:** `brotli.constants.BROTLI_DEFAULT_WINDOW`
+* `lgblock` {integer} (compression only)
+**Default:** `0`
+* `disableLiteralContextModeling` {boolean} (compression only)
+**Default:** `false`
+* `sizeHint` {integer} (compression only)
+**Default:** `0`
+* `npostfix` {integer} (compression only)
+**Default:** `0`
+* `ndirect` {integer} (compression only)
+**Default:** `0`
+* `disableRingBufferReallocation` {boolean} (decompression only)
+**Default:** `false`
+* `largeWindow` {boolean}
+**Default:** `false`
+* `info` {boolean} (If `true`, returns an object with `buffer` and `engine`)
+
+See [`brotli/encode.h`][] and [`brotli/decode.h`][] for more information on
+these.
+
+## Class: brotli.Brotli
+
+Not exported by the `brotli` module. It is documented here because it is the
+base class of the compressor/decompressor classes.
+
+### brotli.bytesWritten
+
+* {number}
+
+The `brotli.bytesWritten` property specifies the number of bytes written to
+the engine, before the bytes are processed (compressed or decompressed,
+as appropriate for the derived class).
+
+### brotli.close([callback])
+
```

Comment:
It seems we should document types of parameters formally, even of self-evident ones. At least:
```md
* `callback` {Function}
```
Ditto for other cases below.